### PR TITLE
[release/3.1] Fix Component Governance bugs

### PR DESCRIPTION
**Pull request by wtgodbe**

### CODE REVIEW

File: src/DependencyInjection/DI.External.Tests/test/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <TargetFrameworks>$(DefaultNetCoreTargetFramework);net472</TargetFrameworks>
     <RootNamespace>Microsoft.Extensions.DependencyInjection</RootNamespace>
     <SignAssembly>false</SignAssembly>
+    <DisablePackageReferenceRestrictions>true</DisablePackageReferenceRestrictions>
```
Comment (dougbu) [inline src/DependencyInjection/DI.External.Tests/test/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:7]: Prefer `%(AllowExplicitReference)` metadata on the specific reference if that's supported in this repo

File: src/DependencyInjection/DI.External.Tests/test/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj [hunk 2]
```diff
@@ -21,6 +22,8 @@
     <Reference Include="Grace.DependencyInjection.Extensions" />
     <Reference Include="Stashbox.Extensions.Dependencyinjection" />
     <Reference Include="Unity.Microsoft.DependencyInjection" />
+    
+    <PackageReference Include="System.Text.Encodings.Web" Version="4.5.1" />
```
Comment (dougbu) [inline src/DependencyInjection/DI.External.Tests/test/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:26]: Why can't this project use a `@(Reference)` item like others in this PR❔

**Review by dougbu — COMMENTED** [review]
Questions before signoff&hellip;

File: src/DependencyInjection/DI.External.Tests/test/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj (hunk 2 above)
Comment (wtgodbe) [reply src/DependencyInjection/DI.External.Tests/test/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:26]: This project uses the 2.1 era package brought in by `Unity.Microsoft.DependencyInjection` & others, upgrading it to one in the same band seemed like the minimal change

File: src/DependencyInjection/DI.External.Tests/test/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj (hunk 1 above)
Comment (wtgodbe) [reply src/DependencyInjection/DI.External.Tests/test/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj]: Done

**Review by dougbu — APPROVED** [review]

**Pull request merged**